[STC6] remove mapped mass

damienmarchal opened the pull request:
______________________________________________________
<!--- Please leave this at the end of your message -->
This PR: 
- [ ] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [ ] does not generate new unit test failures.
- [ ] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [ ] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

damienmarchal committed:
[SofaKernel] Remove SOFA_SUPPORT_MAPPED_MASS.

The mapped masses does not seems to be used at all since years.
I think it is dating from SOFA_DEV. There is no way to
activate/deactivate it from CMakeLists.txt.

In case this is still an active topic. Please open an issue/PR in case
this is still usefull to you.

damienmarchal committed:
[SofaKernel] CLEAN commented code (non-breaking-at-all)

damienmarchal commented:
@guparan Could we focus on having the stc5 & stc6 merged because it is very annoying to have worked and see no results.

guparan commented:
[ci-build][with-scene-tests]

guparan commented:
@damienmarchal Could you take a look at the 8 failing scene tests?
All of them are "required data" errors.

damienmarchal commented:
Sure...

I split the PR by removing everything that was not related to the removable of mapped mass. But I still have no idea why the required data are failing know.

damienmarchal commented:
Actually the failure are generated by error messages like: 
[ERROR]   [BeamFEMForceField(FEM)] Required data "poissonRatio" has not been set. (Current value is 0.49)

My first comment is that these message are 'wrong' because they are "errors" while there is a default value provided to the field. So because of these default value the fields shouldn't be taggued as "required".

damienmarchal commented on SofaKernel/framework/sofa/core/objectmodel/BaseObject.cpp:
```diff
@@ -244,36 +238,22 @@ void BaseObject::copyAspect(int destAspect, int srcAspect)
 		if ((*iData)->isRequired() && !(*iData)->isSet())
 		{
-            serr << "Required data \"" << (*iData)->getName() << "\" has not been set. (Current value is " << (*iData)->getValueString() << ")" << sendl;
+            msg_error() << "Required data \"" << (*iData)->getName() << "\" has not been set. (Current value is " << (*iData)->getValueString() << ")" ;
```
Here is the problematic line that cause scene errors. 
in previous code 'serr' is actually emitting a warning (I already told all the bad I think about that ;))
I replaced that with msg_error which is wrong.

damienmarchal committed:
[SofaKernel] RESTORE old behavior  msg_error() vs msg_warning()

I restored the old behavior until we find a definitive design to expersss the
component requirement and how they should react to this.

damienmarchal commented:
Done...can you review and merge :)

EDIT: actually I put everything in PR #831
